src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -1058,7 +1058,8 @@ void CompileBroker::possibly_add_compiler_threads(JavaThread* THREAD) {
   if (new_c2_count <= old_c2_count && new_c1_count <= old_c1_count) return;
 
   // Now, we do the more expensive operations.
-  julong free_memory = os::free_memory();
+  size_t free_memory = 0;
+  os::free_memory(free_memory);
```

Comment:
I wonder if this should be changed to something like this:
```suggestion
  // Return value ignored - defaulting to 0 on failure.
  (void)os::free_memory(free_memory);
```
to clearly indicate the intent to skip the error check?

Reply:
This is a valid point, but then we'll need to address it somehow else once a pattern with `[[nodiscard]]` or enum class return value is introduced. Or, simply add an empty clause for failure. For now, I will just put a comment there as suggested.

Reply:
Note that the main point here was the cast to (void) used to indicate that the return value is explicitly discarded. Does that not work with the [[nodiscard]]?

Reply:
Sorry, I completely missed that point. Need to check this.

Reply:
Explicitly casting to void is the canonical idiom for ignoring [[nodiscard]].